hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndex.java

```diff
@@ -161,24 +209,43 @@ private Connection getHBaseConnection() {
       }
       return taggedRecords.iterator();
     }
+
+    private Get generateStatement(String key) throws IOException {
+      return new Get(Bytes.toBytes(key)).setMaxVersions(1)
+          .addColumn(SYSTEM_COLUMN_FAMILY, COMMIT_TS_COLUMN)
+          .addColumn(SYSTEM_COLUMN_FAMILY, FILE_NAME_COLUMN)
+          .addColumn(SYSTEM_COLUMN_FAMILY, PARTITION_PATH_COLUMN);
+    }
+
+    private boolean checkIfValidCommit(String commitTs) {
+      HoodieTimeline commitTimeline = hoodieTable.getCompletedCommitTimeline();
+      // Check if the last commit ts for this row is 1) present in the timeline or
+      // 2) is less than the first commit ts in the timeline
+      return !commitTimeline.empty() && (commitTimeline.containsInstant(
```

Comment:
you need the empty check explicitly? wont't containsInstant handle that for you?

Reply:
Good point.